lib/sparkle_formation/utils.rb

```diff
@@ -3,6 +3,43 @@ class SparkleFormation
   # Helper utilities
   module Utils
 
+    module TypeCheckers
+
+      # Validate given value is type defined within valid types
+      #
+      # @param val [Object] value
+      # @param types [Class, Array<Class>] valid types
+      # @return [NilClass]
+      # @raises [TypeError]
+      def __t_check(val, types)
+        types = [types] unless types.is_a?(Array)
+        if(types.none?{|t| val.is_a?(t)})
```

Comment:
never saw a ruby project us this if style ... `if types.none?{|t| val.is_a?(t)}` would be more common